README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

[![Build
Status](http://badges.herokuapp.com/travis/mdsumner/ozmaps?branch=master&env=BUILD_NAME=trusty_release&label=ubuntu)](https://travis-ci.org/mdsumner/ozmaps)
[![Build
Status](http://badges.herokuapp.com/travis/mdsumner/ozmaps?branch=master&env=BUILD_NAME=osx_release&label=osx)](https://travis-ci.org/mdsumner/ozmaps)
[![AppVeyor Build
Status](https://ci.appveyor.com/api/projects/status/github/mdsumner/ozmaps?branch=master&svg=true)](https://ci.appveyor.com/project/mdsumner/ozmaps)
[![CRAN
status](http://www.r-pkg.org/badges/version/ozmaps)](https://cran.r-project.org/package=ozmaps)
![cranlogs](http://cranlogs.r-pkg.org./badges/ozmaps) [![Coverage
Status](https://img.shields.io/codecov/c/github/mdsumner/ozmaps/master.svg)](https://codecov.io/github/mdsumner/ozmaps?branch=master)

# Overview

The goal of ozmaps is to get maps of Australia to plot\!

# Installation

Install from CRAN with

``` r
install.packages("ozmaps")
```

The development version of ozmaps may be installed directly from github.

``` r
devtools::install_github("mdsumner/ozmaps")
```

The package includes some *simple features* data, which can be used
independently of ozmaps with the `sf` package. If required, install `sf`
from CRAN.

**NOTE:** Since April 2020) we *must* have sf installed because of
requirements of the tibble package.

If sf causes you problems or you can’t work with it [get in
touch](https://github.com/mdsumner/ozmaps/issues/) and I will help you.

``` r
install.packages("sf")
```

## Usage

Plot Australia with states.

``` r
library(ozmaps)
ozmap()
```

<img src="man/figures/README-ozmap-1.png" width="100%" />

Plot Australia without states.

``` r
ozmap(x = "country")
```

<img src="man/figures/README-country-1.png" width="100%" />

Add to an existing plot.

``` r
plot(quakes[c("long", "lat")], xlim = c(120, 190))
ozmap(add = TRUE)
```

<img src="man/figures/README-add-plot-1.png" width="100%" />

Obtain the data used in `sf` form.

``` r
sf_oz <- ozmap_data("states")

tibble::as_tibble(sf_oz)
#> # A tibble: 9 x 2
#>   NAME                                                                  geometry
#>   <chr>                                                       <MULTIPOLYGON [°]>
#> 1 New South Wales      (((150.7016 -35.12286, 150.6611 -35.11782, 150.6373 -35.…
#> 2 Victoria             (((146.6196 -38.70196, 146.6721 -38.70259, 146.6744 -38.…
#> 3 Queensland           (((148.8473 -20.3457, 148.8722 -20.37575, 148.8515 -20.3…
#> 4 South Australia      (((137.3481 -34.48242, 137.3749 -34.46885, 137.3805 -34.…
#> 5 Western Australia    (((126.3868 -14.01168, 126.3625 -13.98264, 126.3765 -13.…
#> 6 Tasmania             (((147.8397 -40.29844, 147.8902 -40.30258, 147.8812 -40.…
#> 7 Northern Territory   (((136.3669 -13.84237, 136.3339 -13.83922, 136.3532 -13.…
#> 8 Australian Capital … (((149.2317 -35.222, 149.2346 -35.24047, 149.2716 -35.27…
#> 9 Other Territories    (((167.9333 -29.05421, 167.9188 -29.0344, 167.9313 -29.0…
```

Plot with a custom palette.

``` r
library(sf)
if (utils::packageVersion("paletteer") < '1.0.0') {
 pal <- paletteer::paletteer_d(package = "ochRe", palette = "namatjira_qual")
} else {
  pal <- paletteer::paletteer_d(palette = "ochRe::namatjira_qual")
} 

opal <- colorRampPalette(pal)
nmjr <- opal(nrow(sf_oz))
plot(st_geometry(sf_oz), col = nmjr)
```

<img src="man/figures/README-ochRe-1.png" width="100%" />

``` r

## plot directly with ggplot2
library(ggplot2)
ggplot(sf_oz, aes(fill = NAME)) + geom_sf() + coord_sf(crs = "+proj=lcc +lon_0=135 +lat_0=-30 +lat_1=-10 +lat_2=-45 +datum=WGS84") + scale_fill_manual(values = nmjr)
```

<img src="man/figures/README-ochRe-2.png" width="100%" />

Plot the ABS layers (from 2016).

``` r
ozmap("abs_ced", col = opal(nrow(abs_ced)))
```

<img src="man/figures/README-abs-1.png" width="100%" />

``` r


ozmap("abs_ste", col = opal(nrow(abs_ste)))
```

<img src="man/figures/README-abs-2.png" width="100%" />

## Resolution

These ABS layers `abs_ced`, `abs_lga`, and `abs_ste` are derived from
the 2016 sources and simplified using `rmapshaper::ms_simplify(, keep
= 0.05, keep_shapes = TRUE)` so all the original polygons are there.
There is sufficient detail to map many (most?) of the regions on their
own, which was a major goal for this package.

The cache of the source data at original resolution is available in
[ozmaps.data](https://github.com/mdsumner/ozmaps.data/).

Compare the detail of Bruny Island here in this box, compared with the
very basic maps package layer.

``` r
library(dplyr)
kbor <- abs_lga %>% dplyr::filter(grepl("Kingborough", NAME))
bb <- st_bbox(kbor)

layout(matrix(c(1, 1, 1, 2, 2, 2, 2, 2, 2), nrow = 3))
plot(kbor, reset = FALSE, main = "Kingborough (TAS)")
rect(bb["xmin"], bb["ymin"], bb["xmax"], bb["ymax"])
library(mapdata)
#> Loading required package: maps
par(mar = rep(0, 4))
plot(c(145, 148.5), c(-43.6, -40.8), type = "n", asp = 1/cos(mean(bb[c(2, 4)]) * pi/180), axes = FALSE, xlab = "", ylab = "")
maps::map(database = "worldHires", regions = "australia", xlim = c(145, 148.5), ylim = c(-43.6, -40.8), add = TRUE)
rect(bb["xmin"], bb["ymin"], bb["xmax"], bb["ymax"])
```

<img src="man/figures/README-detail-1.png" width="100%" />

-----

Please note that the ‘ozmaps’ project is released with a [Contributor
Code of
Conduct](https://github.com/mdsumner/ozmaps/blob/master/CODE_OF_CONDUCT.md).
By contributing to this project, you agree to abide by its terms.


NEWS.md
# ozmaps dev

* Avoid sf messages, update projection format. Thanks Edzer Pebesma https://github.com/mdsumner/ozmaps/issues/13 .


# ozmaps 0.4.0

* Now using LazyDataCompression xz. 

* Removed names from abs layer geometries see #3, thanks to @AltfunsMA 

# ozmaps 0.3.6

* Fixed large pdf artefact in tests. 

* ozmaps now imports sf because changes in tibble 3.0.0 require that
`vec_proxy.sfc()` be available (introduced to sf in November 2019). Thanks to @krlmlr. 

If sf causes
you problems or you can't work with it get in touch and I will help you. 

* README improvement thanks to Darío Hereñú (@kant). 

# ozmaps 0.3.0

* Updated examples to work with paletteer 1.0.0


R/ozmaps-package.R
#' @keywords internal
"_PACKAGE"

globalVariables(c("ozmap_country", "ozmap_states",
                  "abs_ced", "abs_lga",   "abs_ste"))


#' Australian map data
#'
#' Australian coastline and boundaries data, including states and territories
#'
#' In-built data set of Australian coastline and provinces (states and
#' territories) simplified from the Australian Bureau of Statistics layer
#' [abs_ste].
#' @name ozmap_states
#' @aliases ozmap_country
#' @docType data
NULL




# this documentation was copied directly from ozmaps.data 0.0.1

#' Australian Bureau of Statistics (ABS) map data
#'
#' These data sets are simplifications of the formal statistical areas
#' and regions published by the ABS in 2016.
#'
#' Each layer was read from the source file with 'sf' package and attributes were removed, leaving a single 'NAME'
#' column from the year-specific column names. The geometry has been
#' simplified using 'rmapshaper::ms_simplify' with default arguments (0.05 detail).
#'
#' Several layers are not included from the total available.
#'
#' The entire nation layer 'AUST' is not included
#' as it is the union of the State and Territory layer.
#'
#' Statistical Areas Level 1 is not included as it is very large (56Mb after simplification).
#'
#' The mesh blocks are not included, nor Greater Capital City Statistical Areas,
#' Indigenous Regions, Remoteness Structure, Statistical Areas (L2, L3, L4),
#' State Electoral Divisions. See 'ozmaps.data' for these.
#'
#'
#' @section Data layers:
#' \describe{
##' \item{abs_ced}{Commonwealth Electoral Divisions}
#' \item{abs_lga}{Local Government Areas}
#' \item{abs_ste}{State and Territory}

#' }
#' @seealso The script to create the data set: data-raw/abs-inbuilt.R
#'
#' @format Simple features data frame with
#' \describe{
#' \item{NAME}{Area name}
#' \item{geometry}{Geometry column in 'sfc' format}
#' }
#'
#' @examples
#' ozmap("abs_ste")
#'
#' ozmap("abs_lga", col = sample(rainbow(nrow(abs_lga), alpha = .4)))
#' pal <- rainbow(12, alpha = 0.6)  ## boring!  install paletteer for ochRe palettes
#'
#' if (isTRUE(requireNamespace("paletteer", quietly = TRUE))) {
#'   if (utils::packageVersion("paletteer") < '1.0.0') {
#'    pal <- paletteer::paletteer_d(package = "ochRe", palette = "namatjira_qual")
#'  } else {
#'    pal <- paletteer::paletteer_d(palette = "ochRe::namatjira_qual")
#'  }
#' }
#' opal <- colorRampPalette(pal)
#' ozmap("abs_ced", col = opal(30))
#' @aliases abs_ced abs_lga abs_ste
#' @name abs-data
NULL
